### DISCUSSION TITLE
Allow differing collision layers and masks for different collision shape nodes

### DISCUSSION THREAD

**Discussion by secretmeister2** [godotengine/godot-proposals#11036] [2D]
It would be useful if the collision layer and mask selection was per collision shape?polygon instead of per object, because you might want different solid collision boxes for different collision surfaces. 

The usecase that I would immediately use this for would be different hitbox for vertical oneways in a sidescroller, as my normal hitbox has a slant, that makes oneways really annoying. If I could have a tiny little hitbox at the feet that was the only thing that collided with oneways, that would fix it all.

**Comment by AThousandShips** [comment]
Thank you for your proposal, please see:
* https://github.com/godotengine/godot-proposals/issues/7400

Closing because there is already a proposal open for this!
